ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/execution/InsertValuesExecutor.java

```diff
@@ -348,6 +352,21 @@ private static void ensureKeySchemasMatch(
     }
 
     if (latest.isPresent() && !latest.get().getSchema().equals(schema.canonicalString())) {
+      // Hack: skip comparing connect name. See https://github.com/confluentinc/ksql/issues/7211
+      // Avro schema are registered in source creation time as well data insertion time.
+      // CONNECT_META_DATA_CONFIG is configured to false in Avro Serializer, but it's true in
+      // AvroSchemaTranslator. It needs to be true to make ConnectSchema map type work. But
+      // enabling it breaks lots of history QTT test which implies backward compatibility issues.
```

Comment:
we should verify that this actually causes compatibility issues before committing a "hack". if it isn't actually a compatibility issue we should just rewrite the old files.

Reply:
If `connect` meta is enabled during serialization, output schema will have connect meta. QTT test failures are from historical ones like: https://github.com/confluentinc/ksql/blob/master/ksqldb-functional-tests%2Fsrc%2Ftest%2Fresources%2Fhistorical_plans%2Fdecimal_-_mod%2F5.5.0_1581572087725%2Fspec.json#L186-L209

New schema for this case will be:
```
{
    "name":"TEST2",
    "keyFormat":{
        "format":"KAFKA"
    },
    "valueFormat":{
        "format":"AVRO"
    },
    "partitions":4,
    "valueSchema":{
        "type":"record",
        "name":"KsqlDataSourceSchema",
        "namespace":"io.confluent.ksql.avro_schemas",
        "fields":[
            {
                "name":"RESULT",
                "type":[
                    "null",
                    {
                        "type":"bytes",
                        "scale":2,
                        "precision":4,
                        "connect.version":1,
                        "connect.parameters":{
                            "scale":"2",
                            "connect.decimal.precision":"4"
                        },
                        "connect.name":"org.apache.kafka.connect.data.Decimal",
                        "logicalType":"decimal"
                    }
                ],
                "default":null
            }
        ],
        "connect.name":"io.confluent.ksql.avro_schemas.KsqlDataSourceSchema"
    }
}
```

This one seems fine but there are ~1000 QTT failures and I'm not sure if they are all ok... What's the best way to double check and if we are going to rewrite old files, how to do that?

Reply:
Is the only change you're expecting to see the addition of the `connect.name` field in Avro schemas? What about the other metadata mentioned in the [docs for `connect.meta.data`](https://docs.confluent.io/platform/current/schema-registry/connect.html) such as doc, version, type, etc? Are those unchanged because ksql doesn't use them?